src/Foundation/NSObject2.cs

```diff
@@ -734,26 +734,17 @@ public override int GetHashCode ()
 
 		public override bool Equals (object obj)
 		{
-			if (obj == null)
-				return false;
 			var o = obj as NSObject;
 			if (o == null)
 				return false;
-			// is only one is a direct binding then both cannot be equals
-			if (IsDirectBinding != o.IsDirectBinding)
-				return false;
+				
 			// we can only ask `isEqual:` to test equality if both objects are direct bindings
-			return IsDirectBinding ? IsEqual (o) : Object.ReferenceEquals (this, obj);
+			bool isDirectBinding = IsDirectBinding;
+			return (isDirectBinding == o.IsDirectBinding) && (isDirectBinding ? IsEqual (o) : ReferenceEquals (this, o));
```

Comment:
The IsDirectBinding equality check is here.

Reply:
yes, it just a readability comment :)

Reply:
Will fix!